contracts/p1/StRSRVotes.sol

```diff
@@ -73,36 +86,39 @@ contract StRSRP1Votes is StRSRP1, IStRSRVotes {
         return pos == 0 ? 0 : _checkpoints[era][account][pos - 1].val;
     }
 
-    function getPastVotes(address account, uint256 blockNumber) public view returns (uint256) {
-        require(blockNumber < NetworkConfigLib.blockNumber(), "ERC20Votes: block not yet mined");
-        uint256 pastEra = _checkpointsLookup(_eras, blockNumber);
-        return _checkpointsLookup(_checkpoints[pastEra][account], blockNumber);
+    function getPastVotes(address account, uint256 timepoint) public view returns (uint256) {
+        require(timepoint < block.timestamp, "ERC20Votes: future lookup");
+
+        uint256 pastEra = _checkpointsLookup(_eras, timepoint);
+        return _checkpointsLookup(_checkpoints[pastEra][account], timepoint);
     }
 
-    function getPastTotalSupply(uint256 blockNumber) public view returns (uint256) {
-        require(blockNumber < NetworkConfigLib.blockNumber(), "ERC20Votes: block not yet mined");
-        uint256 pastEra = _checkpointsLookup(_eras, blockNumber);
-        return _checkpointsLookup(_totalSupplyCheckpoints[pastEra], blockNumber);
+    function getPastTotalSupply(uint256 timepoint) public view returns (uint256) {
+        require(timepoint < block.timestamp, "ERC20Votes: future lookup");
+
+        uint256 pastEra = _checkpointsLookup(_eras, timepoint);
+        return _checkpointsLookup(_totalSupplyCheckpoints[pastEra], timepoint);
     }
 
-    function getPastEra(uint256 blockNumber) public view returns (uint256) {
-        require(blockNumber < NetworkConfigLib.blockNumber(), "ERC20Votes: block not yet mined");
-        return _checkpointsLookup(_eras, blockNumber);
+    function getPastEra(uint256 timepoint) public view returns (uint256) {
+        require(timepoint < block.timestamp, "ERC20Votes: future lookup");
+
+        return _checkpointsLookup(_eras, timepoint);
     }
 
     /// Return the value from history `ckpts` that was current for block number `blockNumber`
-    function _checkpointsLookup(Checkpoint[] storage ckpts, uint256 blockNumber)
+    function _checkpointsLookup(Checkpoint[] storage ckpts, uint256 timepoint)
         private
         view
         returns (uint256)
     {
         // We run a binary search to set `high` to the index of the earliest checkpoint
-        // taken after blockNumber, or ckpts.length if no checkpoint was taken after blockNumber
+        // taken after timepoint, or ckpts.length if no checkpoint was taken after timepoint
         uint256 high = ckpts.length;
         uint256 low = 0;
         while (low < high) {
             uint256 mid = MathUpgradeable.average(low, high);
-            if (ckpts[mid].fromBlock > blockNumber) {
+            if (ckpts[mid].fromBlock > timepoint) {
```

Comment:
probably `fromBlock` should be renamed in the Checkpoint struct and reflect that is time based?

Reply:
I did think about it, I'm open to it. I think OZ Upgrades might complain a lot for that one.

Reply:
Ok I thought it was our own structure not OZ